lib/spack/spack/config.py

```diff
@@ -763,57 +764,93 @@ def _merge_yaml(dest, source):
     with `::` instead of `:`, and the key will override that of the
     parent instead of merging.
 
+    If override is True, then source will be copied to dest unconditionally.
     """
     def they_are(t):
         return isinstance(dest, t) and isinstance(source, t)
 
-    # If both are None, handle specially and return None.
-    if source is None and dest is None:
+    if source is None:
```

Comment:
Nope. If `source` is `None` then the values of `dest` and `override` are both irrelevant. If `source` and `dest` are both `None`,  the behavior is unchanged. `dest` is not `None`, then even if  `override == False` in the new code we would still drop through to `copy.copy(source)`. If `override == True` you'd just be returning `None` instead of the result of `copy.copy(None)`. In the light of your comment on the redundancy of the `override` flag however, this comment is redundant.